addons/stock/views/stock_picking_views.xml

```diff
@@ -127,9 +127,12 @@
                 <header>
                     <button name="action_confirm" attrs="{'invisible': [('state', '!=', 'draft')]}" string="Mark as Todo" type="object" class="oe_highlight" groups="base.group_user" data-hotkey="x"/>
                     <button name="action_assign" attrs="{'invisible': [('show_check_availability', '=', False)]}" string="Check Availability" type="object" class="oe_highlight" groups="base.group_user" data-hotkey="q"/>
-                    <button name="button_validate" attrs="{'invisible': [('state', 'in', ('waiting','confirmed', 'draft', 'done', 'cancel'))]}" string="Validate" type="object" class="oe_highlight" groups="stock.group_stock_user" data-hotkey="v"/>
-                    <button name="button_validate" attrs="{'invisible': [('state', 'in', ('draft', 'assigned', 'done', 'cancel'))]}" string="Validate" type="object" groups="stock.group_stock_user" class="o_btn_validate" data-hotkey="v"/>
-                    <button name="action_reset_draft" attrs="{'invisible': [('state', 'in', ('draft', 'done', 'cancel'))]}" string="Set Draft" type="object" groups="stock.group_stock_user" class="o_highlight" data-hotkey="d"/>
+                    <button name="button_validate" attrs="{'invisible': ['|', ('picking_type_code', '!=', 'outgoing'), ('state', 'in', ('waiting', 'confirmed', 'draft', 'done', 'cancel'))]}" string="Delivered" type="object" class="oe_highlight" groups="stock.group_stock_user" data-hotkey="v"/>
+                    <button name="button_validate" attrs="{'invisible': ['|', ('picking_type_code', '!=', 'outgoing'), ('state', 'in', ('draft', 'assigned', 'done', 'cancel'))]}" string="Delivered" type="object" groups="stock.group_stock_user" class="o_btn_validate" data-hotkey="v"/>
+                    <button name="button_validate" attrs="{'invisible': ['|', ('picking_type_code', '!=', 'internal'), ('state', 'in', ('waiting', 'confirmed', 'draft', 'done', 'cancel'))]}" string="Transferred" type="object" class="oe_highlight" groups="stock.group_stock_user" data-hotkey="v"/>
+                    <button name="button_validate" attrs="{'invisible': ['|', ('picking_type_code', '!=', 'internal'), ('state', 'in', ('draft', 'assigned', 'done', 'cancel'))]}" string="Transferred" type="object" groups="stock.group_stock_user" class="o_btn_validate" data-hotkey="v"/>
+                    <button name="button_validate" attrs="{'invisible': ['|', ('picking_type_code', '!=', 'incoming'), ('state', 'in', ('waiting', 'confirmed', 'draft', 'done', 'cancel'))]}" string="Received" type="object" class="oe_highlight" groups="stock.group_stock_user" data-hotkey="v"/>
+                    <button name="button_validate" attrs="{'invisible': ['|', ('picking_type_code', '!=', 'incoming'), ('state', 'in', ('draft', 'assigned', 'done', 'cancel'))]}" string="Received" type="object" groups="stock.group_stock_user" class="o_btn_validate" data-hotkey="v"/>
```

Comment:
You can't do a widget that is base on a computed field to avoid that. Seems super complex and not scalable

Reply:
If it's too difficult/complex -> skip

Reply:
@amoyaux 

can you suggest a better approach?

Reply:
> If it's too difficult/complex -> skip

@amoyaux can we skip for now?

Reply:
Yes